recipes/zlib-bytesread-to-byteswritten/src/workflow.ts

```diff
@@ -0,0 +1,140 @@
+import { getNodeRequireCalls } from "@nodejs/codemod-utils/ast-grep/require-call";
+import { getNodeImportStatements } from "@nodejs/codemod-utils/ast-grep/import-statement";
+import { resolveBindingPath } from "@nodejs/codemod-utils/ast-grep/resolve-binding-path";
+import { removeLines } from "@nodejs/codemod-utils/ast-grep/remove-lines";
+import type { Edit, Range, SgRoot } from "@codemod.com/jssg-types/main";
+import type Js from "@codemod.com/jssg-types/langs/javascript";
+
+const ZLIB_FACTORIES = [
+	"createGzip",
+	"createGunzip",
+	"createDeflate",
+	"createInflate",
+	"createBrotliCompress",
+	"createBrotliDecompress",
+	"createUnzip",
+];
+
+export default function transform(root: SgRoot<Js>): string | null {
+	const rootNode = root.root();
+	const edits: Edit[] = [];
+	const linesToRemove: Range[] = [];
+
+	// 1 Find all static zlib imports/requires
+	const importNodes = [
+		...getNodeRequireCalls(root, "node:zlib"),
+		...getNodeImportStatements(root, "node:zlib")
+	];
+
+	const factoryBindings: string[] = [];
+	const streamVariables: string[] = [];
+
+	for (const node of importNodes) {
+		const baseBind = resolveBindingPath(node, "$");
+
+		if (baseBind) {
+			for (const factory of ZLIB_FACTORIES) factoryBindings.push(`${baseBind}.${factory}`);
+		}
+		for (const factory of ZLIB_FACTORIES) {
+			const binding = resolveBindingPath(node, `$.${factory}`);
+			if (binding && !factoryBindings.includes(binding)) factoryBindings.push(binding);
+		}
```

Comment:
```js
const baseBind = resolveBindingPath(node, "$");
```

I do not think the resolveBindingPath, can works in this way, you need to specify which binding you want to get the path, just '$' not means anything for this function.

So I think you code can be refactored to below one, if I missed something, let me know
```suggestion
		for (const factory of ZLIB_FACTORIES) {
			const binding = resolveBindingPath(node, `$.{factory}`);
			if (binding && !factoryBindings.includes(binding)) factoryBindings.push(binding);
		}
```